core/store/src/trie/split.rs

```diff
@@ -21,6 +24,24 @@ const MAX_NIBBLES: usize = AccountId::MAX_LEN * 2;
 // not to add any subtrees here that do not use account ID as key (or key prefix).
 const SUBTREES: [u8; 4] = [ACCOUNT, CONTRACT_CODE, ACCESS_KEY, CONTRACT_DATA];
 
+#[derive(Error, Debug)]
+pub enum SplitError {
+    #[error(transparent)]
+    Storage(#[from] StorageError),
+    #[error("no root in trie")]
+    NoRoot,
+    #[error("split not found – trie is empty or contains invalid keys")]
+    SplitNotFound,
+    #[error("key in the trie is not a valid account ID (too short or odd length). nibbles: {0:?}")]
+    InvalidKey(Vec<u8>),
```

Comment:
mini nit: it's a bit inconsistent to add Invalid here where all other errors don't have it.

Reply:
I don't understand the issue. What's your suggestion?

Reply:
To either add `Invalid` prefix to all variants or remove it so that none have it.